app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
+import org.mozilla.fenix.nimbus.StyleData
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
 
+/**
+ * View holder for the Nimbus Message Card.
+ *
+ * @property interactor [SessionControlInteractor] which will have delegated to all user
+ * interactions.
+ */
 class MessageCardViewHolder(
-    view: View,
-    private val interactor: SessionControlInteractor
-) : RecyclerView.ViewHolder(view) {
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: SessionControlInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+
+    private lateinit var messageGlobal: Message
 
     fun bind(message: Message) {
-        val binding = NimbusMessageCardBinding.bind(itemView)
+        messageGlobal = message
+        interactor.onMessageDisplayed(message)
+    }
 
-        if (message.data.title.isNullOrBlank()) {
-            binding.titleText.isVisible = false
-        } else {
-            binding.titleText.text = message.data.title
-        }
+    companion object {
+        internal val LAYOUT_ID = View.generateViewId()
+    }
 
-        binding.descriptionText.text = message.data.text
+    @Composable
+    override fun Content() {
 
-        if (message.data.buttonLabel.isNullOrBlank()) {
-            binding.messageButton.isVisible = false
-            binding.experimentCard.setOnClickListener {
-                interactor.onMessageClicked(message)
-            }
-        } else {
-            binding.messageButton.text = message.data.buttonLabel
-            binding.messageButton.setOnClickListener {
-                interactor.onMessageClicked(message)
+        MessageCardPlaceHolder(
+            message = messageGlobal,
+            onMessageClicked = { interactor.onMessageClicked(messageGlobal) },
+            onMessageClosedClicked = { interactor.onMessageClosedClicked(messageGlobal) }
+        )
+    }
+}
+
+/**
+ * Message Card.
+ *
+ * @param message contains title and description of message.
+ * @param onMessageClicked Invoked when user clicks on the button.
+ * @param onMessageClosedClicked Invoked when user clicks on close button to remove message.
+ */
+@Suppress("LongMethod")
+@Composable
+fun MessageCardPlaceHolder(
+    message: Message,
+    onMessageClicked: () -> Unit,
+    onMessageClosedClicked: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .padding(16.dp)
+            .then(
+                if (message.data.buttonLabel.isNullOrBlank()) {
+                    Modifier.clickable(onClick = onMessageClicked)
+                } else {
+                    Modifier
+                }
+            ),
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            if (!message.data.title.isNullOrBlank()) {
```

Comment:
Could we add the condition to just cover the `SectionHeader` ? As it looks like we are duplicating `IconButton` and `PrimaryText` in both part of the condition.

 
```kotlin
            if (!message.data.title.isNullOrBlank()) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                ) {

                    SectionHeader(
                        text = message.data.title!!,
                        modifier = Modifier
                            .weight(1f)
                            .padding(bottom = 10.dp)
                    )

                    IconButton(
                        modifier = Modifier.size(20.dp),
                        onClick = onMessageClosedClicked
                    ) {
                        Icon(
                            painter = painterResource(R.drawable.ic_close),
                            contentDescription = stringResource(
                                R.string.remove_home_collection_placeholder_content_description
                            ),
                            tint = FirefoxTheme.colors.iconPrimary
                        )
                    }
                }
                PrimaryText(
                    text = message.data.text,
                    modifier = Modifier.fillMaxWidth(),
                    fontSize = 14.sp
                )
            } else {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                ) {
                    PrimaryText(
                        text = message.data.text,
                        modifier = Modifier.weight(1f),
                        fontSize = 14.sp
                    )
                    IconButton(
                        modifier = Modifier.size(24.dp),
                        onClick = onMessageClosedClicked
                    ) {
                        Icon(
                            painter = painterResource(R.drawable.ic_close),
                            contentDescription = stringResource(
                                R.string.remove_home_collection_placeholder_content_description
                            ),
                            tint = FirefoxTheme.colors.iconPrimary
                        )
                    }
                }
            }
```

Reply:
The thing is when title is empty PrimaryText falls within the Row and not otherwise, hence the if condition is outside the Row not SectionHeader

Reply:
Got it 👍🏽 !